### PR TITLE
Prepare v2.0.0

### LINKED ISSUE #37: Add a warning when using too old golangci version
The last release (https://github.com/golangci/golangci-lint-action/pull/34) brakes all the projects using `golangci-lint-action@v1` and `version = v1.27` (easy fix -> bump at `v1.28`).

Example: https://github.com/moul/golang-repo-template/pull/62
  * dummy commit to trigger the CI: https://github.com/moul/golang-repo-template/pull/62/checks?check_run_id=862641477
  * fix https://github.com/moul/golang-repo-template/pull/62/checks?check_run_id=862642684

What do you think about adding a more explicit warning when the user's `version` is older than a `minimal_supported_version` constant you keep updated internally in this repo?

### LINKED ISSUE #39: v1.2.2 was broke if using working-directory
For example: 
lint-go:
    name: Lint Go code
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      - name: Run golangci-lint for dice
        uses: golangci/golangci-lint-action@v1
        with:
          version: v1.26
          working-directory: ./dice
          args: --timeout 5m

Output err:
  Running [/home/runner/golangci-lint-1.26.0-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=./dice --timeout 5m] in [/home/runner/work/aws-solutions-assembler/aws-solutions-assembler/dice] ...
  level=error msg="Can't get config for command line: can't parse args: unknown flag: --path-prefix"

### PR DESCRIPTION
**Pull request by SVilgelm**
Changed the minimum version, In order to fix the compatibility issues of v1.2.2 action and v1.28.3 lint.
Add compatibility section in README.md

Fixes #39 #37

After merging this PR we need to raise 1.2.3 release and remove 1.2.2 release (deprecate it)

### CODE REVIEW

**Comment by SVilgelm** [comment]
@sayboras any thoughts?

File: README.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ The action runs [golangci-lint](https://github.com/golangci/golangci-lint) and r
 
 ![GitHub Annotations](./static/annotations.png)
 
+## Compatibility
+
+* `v1.2.3` works with `golangci-lint` version >= `v1.28.3`
```
Comment (sayboras) [inline README.md:12]: not sure if this is still considered as semantic versioning.

The change we made in 1.2.2 is considered as breaking, so having restriction on patch version (e.g. v1.2.3) is not considered in my opinion as backward compatible.

Comment (SVilgelm) [reply README.md:12]: Is 1.3.0 good option?
Or we should go with 2.0.0?

**Comment by sayboras** [comment]
> @sayboras any thoughts?

@SVilgelm just want to boucing idea here. One alternative to fix this one, and make release backward compatible for patch version is to check `golangci-lint` version, and decide if we need to append the `path-prefix`  or not. What do you think ?

**Comment by SVilgelm** [comment]
@sayboras hmmm, I would prefer to release v2 instead of adding such logic, I don't see any problems with removing 1.2.2 release and releasing it as 2.0.0
so someone will need to get back to v1.2.1, someone will just upgrade to v2

My concern here is if we will add such logic once, then what will stop us to do this again and again? and then we will have lots of `if..else` blocks

let's do it right now

**Comment by sayboras** [comment]
> @sayboras hmmm, I would prefer to release v2 instead of adding such logic, I don't see any problems with removing 1.2.2 release and releasing it as 2.0.0
> so someone will need to get back to v1.2.1, someone will just upgrade to v2
> 
> My concern here is if we will add such logic once, then what will stop us to do this again and again? and then we will have lots of `if..else` blocks
> 
> let's do it right now

Yeah, I don't like that a lot of if else either, makes the code ugly to read and maintain.

v2.0.0 sounds good to me. But will v1.x still be under support? golangci-lint supports last two releases I think

**Review by sayboras — APPROVED** [review]
LGTM :100:

File: README.md (hunk 1 above)
Comment (sayboras) [reply README.md]: I personally think 2.0.0 beter. Not sure what others think.

**Comment by denis-tingajkin** [comment]
Also, this PR closes this https://github.com/golangci/golangci-lint-action/issues/38 :)

**Review by denis-tingaikin — APPROVED** [review]

**Comment by SVilgelm** [comment]
> Also, this PR closes this #38 :)

It's not closes, using a patch part of version nis still not allowed, or I did not understand the problem :)

**Pull request merged**